src/psyclone/psyir/symbols/symbol.py

```diff
@@ -323,5 +408,40 @@ def is_unresolved(self):
         '''
         return isinstance(self._interface, UnresolvedInterface)
 
+    def find_symbol_table(self, node):
+        '''
+        Searches back up the PSyIR tree for the SymbolTable that contains
+        this Symbol.
+
+        :param node: the PSyIR node from which to search.
+        :type node: :py:class:`psyclone.psyir.nodes.Node`
+
+        :returns: the SymbolTable containing this Symbol or None.
+        :rtype: :py:class:`psyclone.psyir.symbols.SymbolTable` or NoneType
+
+        :raises TypeError: if the supplied `node` argument is not a PSyIR Node.
+
+        '''
+        from psyclone.psyir.nodes import Node
```

Comment:
Pylint prefers this import to be in the top-level.

There are similar ones in the node::scope and datasymbol::resolve_defered methods touched in this PR.

Reply:
Oh yes, I tried moving that one and got a circular dependence so left it where it was. In fact, that turns out to be the case for all three. I've added comments and disabled the pylint warning in each case.